src/mlio-py/mlio/core/py_device_array.cxx

```diff
@@ -45,9 +45,14 @@ void *py_device_array::make_or_get_string_buffer()
 {
     if (string_buf_.empty()) {
         string_buf_.reserve(span_.size());
-
-        for (py::str obj : span_.as<std::string>()) {
-            string_buf_.push_back(obj.release().ptr());
+        try {
+            for (py::str obj : span_.as<std::string>()) {
+                string_buf_.push_back(obj.release().ptr());
+            }
+        } catch (std::runtime_error &) {
+            PyErr_SetString(PyExc_ValueError, "Unable to convert std::string to py::str. Please "
```

Comment:
This error message is too low-level and cryptic for a scientist or ML engineer. I would write something like: "The string tensor contains an invalid UTF-8 sequence. Please make sure that you specify an explicit text encoding if the text supposed to be in a different encoding."

Reply:
+1 on Can's comment. 
Is "string tensor" high-level enough - i.e. should we just say "input".
Additionally, will this error only show up when UTF-8 is input encoding i.e. if user inputs invalid non-UTF8 encoding, what error is displayed?

Reply:
I think `string tensor` should be fine, given that the user will already be working at the example level at that point. `input` might be too generic in my opinion (as it is only a specific feature that is causing the issue).

You mean if the user passes in invalid encoding param? that is handled by mlio at a lower level. For example:
`mlio.StreamError: An invalid byte sequence encountered while converting from ISO-8859-6 to UTF-8.` if user passes in a valid but wrong encoding.

`OSError: [Errno 22] The ISO-8859-abc encoding is not supported by the platform.: Invalid argument` if a user passes in invalid encodings.